ui-tests/base/com.liferay.ide.functional.modules.ext.base/src/com/liferay/ide/functional/modules/ext/base/NewModulesExtWizardGradleBase.java

```diff
@@ -24,35 +24,134 @@
+		wizardAction.finish();
+
+		jobAction.waitForNoRunningProjectBuildingJobs();
+
+		Assert.assertTrue(
+			viewAction.project.visibleFileTry(
+				liferayWorkspace.getName(), "ext", project.getName(), "src/main/java",
+				"com.liferay.login.web.internal.constants", "LoginPortletKeys.java"));
+	}
+
+	public void addModulesExtWithJSP() {
+		overrideOriginalModule("com.liferay.login.web");
+
+		addOverrideFiles("META-INF", "resources", "configuration.jsp");
 
 		wizardAction.finish();
 
 		jobAction.waitForNoRunningProjectBuildingJobs();
 
-		viewAction.project.openFile(liferayWorkspace.getName(), "ext", project.getName(), "build.gradle");
+		Assert.assertTrue(
+			viewAction.project.visibleFileTry(
+				liferayWorkspace.getName(), "ext", project.getName(), "src/main/resources", "META-INF", "resources",
+				"configuration.jsp"));
+	}
+
+	public void addModulesExtWithLanguageProperties() {
+		overrideOriginalModule("com.liferay.login.web");
+
+		addOverrideFiles("content", "Language.properties");
+
+		wizardAction.finish();
+
+		jobAction.waitForNoRunningProjectBuildingJobs();
+
+		Assert.assertTrue(
+			viewAction.project.visibleFileTry(
+				liferayWorkspace.getName(), "ext", project.getName(), "src/main/resources", "content",
+				"Language.properties"));
+	}
+
+	public void addModulesExtWithManifest() {
+		overrideOriginalModule("com.liferay.login.web");
+
+		addOverrideFiles("META-INF", "MANIFEST.MF");
+
+		wizardAction.finish();
+
+		jobAction.waitForNoRunningProjectBuildingJobs();
 
 		Assert.assertTrue(
-			viewAction.project.visibleFileTry(liferayWorkspace.getName(), "ext", project.getName(), "build.gradle"));
+			viewAction.project.visibleFileTry(
+				liferayWorkspace.getName(), "ext", project.getName(), "src/main/resources", "META-INF", "MANIFEST.MF"));
+	}
+
+	public void addModulesExtWithoutOverrideFiles() {
+		overrideOriginalModule("com.liferay.login.web");
 
-		editorAction.close();
+		wizardAction.finish();
 
-		viewAction.project.closeAndDelete(liferayWorkspace.getName(), "ext");
+		jobAction.waitForNoRunningProjectBuildingJobs();
+	}
+
+	public void addModulesExtWithPortletProperties() {
+		overrideOriginalModule("com.liferay.login.web");
+
+		addOverrideFiles("portlet.properties");
+
+		wizardAction.finish();
+
+		jobAction.waitForNoRunningProjectBuildingJobs();
+
+		Assert.assertTrue(
+			viewAction.project.visibleFileTry(
+				liferayWorkspace.getName(), "ext", project.getName(), "src/main/resources", "portlet.properties"));
+	}
+
+	public void addModulesExtWithResourceAction() {
+		overrideOriginalModule("com.liferay.login.web");
+
+		addOverrideFiles("resource-actions", "default.xml");
+
+		wizardAction.finish();
+
+		jobAction.waitForNoRunningProjectBuildingJobs();
+
+		Assert.assertTrue(
+			viewAction.project.visibleFileTry(
+				liferayWorkspace.getName(), "ext", project.getName(), "src/main/resources", "resource-actions",
+				"default.xml"));
+	}
+
+	public void addOverrideFiles(String... file) {
```

Comment:
should not have method, just write actions in the test case.

Reply:
the new method is just include the common action steps all the cases need. If only writing actions, then all the test cases will have same steps code, is that ok?

Reply:
yes, using less methods will make the test cases more readable